src/processor.cpp

```diff
@@ -0,0 +1,413 @@
+
+    //Use component vector to find out if we have usable items on the tile
+    //TODO: Handle processes with multiple possible inputs, see kiln
+    //TODO: Handle processes requiring multiple input materials
+
+    // Seen the code in keg, I could do interesting stuff by manipulating
+    // which components is at which position in the item stack, could make
+    // multi-input processes more bearable
+    map_stack items = g->m.i_at(examp);
+    bool processed_is_present = false;
+    std::map<itype_id, long> charges;
+    item fuel = item::item(current_processor.fuel_type, int(calendar::turn), 0);
+    for (item i : items) {
+        //Please check to make sure this one is working as intended
+        auto index = std::find( active_process.components.begin(), active_process.components.end(), i.typeId() );
+        if (index == active_process.components.end() && i.typeId() != current_processor.fuel_type) {
+            add_msg(m_bad, _("You find some %s, which can't be used in this process."), i.tname(1, false).c_str());
+            add_msg(_("You must remove it before you proceed."));
+            return;
+        }
+        else if (i.typeId() != current_processor.fuel_type) {
+            charges.insert( std::make_pair( i.typeId() , charges.at( i.typeId() ) + i.charges ) );
+        }
+        else {
+            fuel.charges += i.charges;
+        }
+    }
+    //At this point we only have components and fuel on tile
+    //And we know exact numbers
+    //Can someone check if I'm using iterators right in this case?
+    long minimum = LONG_MAX;
+    for( std::pair<itype_id, long> m : charges ){
+        if( m.second == 0 ){
+            add_msg(m_info, _("There's no %s which is needed for the process"), item::nname( m.first, 1 ).c_str() );
+            return;
+        } else if( m.second < minimum ){
+            minimum = m.second;
+        }
+    }
+    //Need to check if the next line performs integer division
+    minimum = std::min( minimum, fuel.charges / active_process.fuel_intake );
+    if ( minimum <= 0 ) {
+        add_msg(_("The current materials present would go to waste without producing any %s, you need more reagents"), item::nname(active_process.output, 1).c_str() );
+        return;
+    }
+    add_msg( _("This process is ready to be started, and it will produce %d %s"), item::nname(active_process.output, minimum).c_str(), minimum );
+    if( !p.has_charges( "fire" , 1 ) ) {
+        add_msg( _("This process is ready to be started, but you have no fire source to start it.") );
+        return;
+    } else if( !query_yn( _("This process is ready to be started, and it will produce %d %s . %s"),
+                        item::nname(active_process.output, minimum).c_str(), minimum, current_processor.full_message.c_str() ) ) {
+        return;
+    }
+    add_msg( _( current_processor.start_message.c_str() ) );
+    p.use_charges( "fire", 1 );
+
+    //TODO: skill
+
+    for( item i : items ){
```

Comment:
Here the use of copies instead of references is not just code quality issue, but will actually make the code not work at all:
You're changing the birthday of copies of items, then discarding the modified copies.